orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -156,23 +171,64 @@ def eval_step(self, *input, **kwargs):
         Returns:
             ordered :py:obj:`list` with model outputs as described by :py:attr:`.ORTTrainer.model_desc`
         """
-        # Export model to ONNX
-        if self._onnx_model is None:
-            sample_input = self._prepare_model_input(
-                self.model_desc.inputs, None, None, *input, **kwargs)
-            self._init_onnx_model(sample_input)
+        # with model_loss_cls, the last input is label, first output is loss
+        sample_input = self._prepare_model_input(self.model_desc.inputs,
+                                                 None, None, *input, **kwargs)
 
+        if self._onnx_model is None:
+            if self._torch_model is not None:
+                self._init_onnx_model(sample_input)
+            else:
+                raise RuntimeError("Model is uninitialized. Only ONNX and PyTorch models are supported")
+
+        input_desc = self.model_desc.inputs[0:len(sample_input)]
+        output_desc = self.model_desc.outputs
+
+        if not isinstance(sample_input, (list, tuple)):
+            sample_input = (sample_input,)
+
+        run_options = ort.RunOptions()
+        run_options.only_execute_path_to_fetches = True
+        run_options.training_mode = False
+        session_run_results = self._training_session_run_helper(False,
+                                                                sample_input,
+                                                                input_desc,
+                                                                output_desc,
+                                                                self.options.device.id,
```

Comment:
could we access the device id directly in the method without passing as a parameter?

Reply:
yes, we could. good catch!